src/Compilers/CSharp/Portable/Symbols/Source/SourceEventSymbol.cs

```diff
@@ -478,10 +478,15 @@ protected void CheckModifiersAndType(DiagnosticBag diagnostics)
                 // A static member '{0}' cannot be marked as override, virtual, or abstract
                 diagnostics.Add(ErrorCode.ERR_StaticNotVirtual, location, this);
             }
-            else if (IsReadOnly && (IsStatic || HasAssociatedField))
+            else if (IsReadOnly && IsStatic)
```

Comment:
I know you didn't choose how this chain of else-ifs was structured, but while you're here consider refactoring this so it's sane: ie, not checking `IsAbstract` 5 times, etc.

Reply:
I contemplated this for a bit and for now I feel introducing nesting or locals in such a long if-else chain is going to be bad for clarity.

Reply:
I agree with @RikkiGibson